pilot/model/config.go

```diff
@@ -457,7 +486,8 @@ func MatchSource(meta ConfigMeta, source *routing.IstioService, instances []*Ser
 	return false
 }
 
-// SortRouteRules sorts a slice of rules by precedence in a stable manner
+// SortRouteRules sorts a slice of v1alpha1 rules by precedence in a stable manner.
+// non-v1alpha1 rules are sorted low without a guaranteed relative ordering.
```

Comment:
? Its an array right?

Reply:
Most of the time in Go you are technically using [slices](https://blog.golang.org/go-slices-usage-and-internals), not arrays.

From the Go blog:
> Slices are analogous to arrays in other languages, but have some unusual properties.